packages/firestore/src/local/local_documents_view.ts

```diff
@@ -252,4 +264,28 @@ export class LocalDocumentsView {
         return results;
       });
   }
+
+  private getDocumentWithPendingPatches(
+    transaction: PersistenceTransaction,
+    matchingMutationBatches: MutationBatch[],
+    results: DocumentMap
+  ): PersistencePromise<NullableMaybeDocumentMap> {
```

Comment:
Would it make sense for this to return a DocumentMap? It would simplify the code above, but you would need to post-filter in here. Alternatively, you could also assemble the final result map here and make this a `populateDocumentMapWithMissingBaseDocuments()` or something similar. Optional.

Reply:
My understanding is `DocumentMap` being an immutable structure makes it awkward to update from the helper function. I have not tried it..but don't expect it to work without extra tricks.

Reply:
You can get return the combined set of `existingDocuments` and the result of this function. The goal would be to simplify the `getDocumentsMatchingCollectionQuery` function.

Optional.

Reply:
Decide not to do it because it is not a significant improvement..